Controllers/DepartmentsController.cs

```diff
@@ -0,0 +1,89 @@
+    {
+        private readonly DepartmentService _departmentService;
+        private readonly EmployeeService _employeeService;
+
+        public DepartmentsController(DepartmentService departmentService, EmployeeService employeeService)
+        {
+            _departmentService = departmentService;
+            _employeeService = employeeService;
+        }
+
+        [HttpPost]
+        [Route("/department")]
+        public async Task<ActionResult<Department>> CreateDepartment([FromBody] Department body)
+        {
+            if (IsValidDepartment(body))
+            {
+                var newDepartment = new Department { ManagerId = body.ManagerId, Name = body.Name};
+
+                var response = await _departmentService.CreateDepartment(newDepartment);
+
+                if (response == null) { return BadRequest(); }
+                else { return Created(String.Empty, response); }
+            }
+
+            return BadRequest("Given object is not valid");
+        }
+
+        [HttpDelete]
+        [Route("/department/{departmentId}")]
+        public async Task<ActionResult> DeleteDepartment([FromRoute][Required] long departmentId)
+        {
+            var response = await _departmentService.DeleteDepartment(departmentId);
+
+            if (response == null) { return NotFound(); }
+            return NoContent();
+        }
+
+        [HttpPut]
+        [Route("/department/{departmentId}")]
+        public async Task<ActionResult<Department>> EditDepartment([FromBody] Department body, [FromRoute][Required] long departmentId)
+        {
+            if (body == null) { return BadRequest(); }
+            var response = await _departmentService.EditDepartment(body, departmentId);
+
+            if (response == null) { return BadRequest(); }
+            else { return Ok(response); }
+        }
+
+        [HttpGet]
+        [Route("/department/{departmentId}")]
+        public async Task<ActionResult<Department>> GetDepartment([FromRoute][Required] long departmentId)
+        {
+            var response = await _departmentService.GetDepartment(departmentId);
+
+            if (response == null) { return NotFound(); }
+            else { return Ok(response); }
+        }
+
+        [HttpGet]
+        [Route("/department")]
+        public async Task<ActionResult<List<Department>>> GetDepartments([FromQuery] int top, [FromQuery] int skip)
+        {
+            if (top < 0 || skip < 0) { return BadRequest(); }
+
+            var response = (top > 0 || skip > 0) ? await _departmentService.GetDepartments(top, skip) : await _departmentService.GetDepartments();
+
+            if (response == null) { return NotFound(); }
+            else { return Ok(response); }
+        }
+
+        public bool IsValidDepartment(Department department)
+        {
+            if (department == null ||
+               String.IsNullOrEmpty(department.Name) ||
+               department.ManagerId == 0) { return false; }
```

Comment:
Remove check for ```department.ManagerId == 0``` it doesn't let create a department without a manager, but employees cannot be created without ```departmentId``` making it impossible to create a department

Reply:
done